### DISCUSSION TITLE
Make window / tab titles aware of job control

### DISCUSSION THREAD

**Discussion by clarkema** [ghostty-org/ghostty#3961] [Feature Requests, Ideas]
This observation is based on macOS 15.2 with zsh.  I imagine it would be the same on other platforms.

The name of the running command in the title is lost if the command is suspended and later resumed using the shell's job control features. For example:

  1. Edit a file with `vim foo.txt`
  2. The window title will show `vim foo.txt`
  3. Suspend vim and return to the shell with `^z`
  4. Window title shows the current path
  5. Resume vim with `fg`
  6. The window title shows `fg`

iTerm seems to do the right thing here (although ghostty is better in that it shows the full command; i.e., `vim foo.txt` in the title, whereas iTerm just shows `vim`.)  I haven't inspected its shell integration, but maybe it's possible to include a special case to check whether the command to be run is `fg` and then look up the name of process to be resumed using `jobs`?

**Comment by 00-kat** [comment]
I can reproduce this on **Bash** and **Zsh** but **not Fish**.  Running on Linux.

**Comment by taylrfnt** [comment]
Just gave this a whirl on one of my test machines:
- macOS 15.3.1 
- ghostty on tip (1.1.3-main+38908e01)
- zsh (5.9 (arm64-apple-darwin24.0))

Using vanilla Vim, I don't see the title change at all.  Using Neovim, I see the title change when editing a file, but it returns properly 
when using `fg` to resume the job.

I don't have a bash or fish install to use (this build is locked into a limited set of packages for some other work I do on it).  I can try and test bash on my other machine at home tonight.

Edit: here's a gif, had to get ffmpeg back on my system 
![shell-suspend-zsh](https://github.com/user-attachments/assets/a1b69c70-d2d5-4ee9-ba4a-f9e3ad27e02d)

**Reply by mkrieger1** [reply]
With Neovim this works because Neovim sets the window title (if `set title` is used).

**Reply by taylrfnt** [reply]
Ah, good callout.  After testing on my home machine, I can confirm that using plain old vim does generate an incorrect window title when resumed in both bash and zsh.  I don't know why my other build doesn't generate a title for vim, but I will chalk that up to some other niche config I have on that build.  

![incorrect-fg-title](https://github.com/user-attachments/assets/9cddd8e5-3577-4b87-ae7f-a2cb2b650865)

I am looking into fish because I just don't use it, but I assume it'll likely align with Kat's results above.

**Reply by 00-kat** [reply]
>  I don't know why my other build doesn't generate a title for vim

Do make sure that [shell integration actually worked].

[shell integration actually worked]: https://ghostty.org/docs/features/shell-integration#verify-shell-integration

**Reply by taylrfnt** [reply]
That's the one.  I disabled it for some testing and forgot :)
